test/Microsoft.AspNet.Mvc.DataAnnotations.Test/RangeAttributeAdapterTest.cs

```diff
@@ -18,6 +20,15 @@ public void GetClientValidationRules_ReturnsValidationParameters()
             var metadata = provider.GetMetadataForProperty(typeof(string), "Length");
 
             var attribute = new RangeAttribute(typeof(decimal), "0", "100");
+            attribute.ErrorMessage = "The field Length must be between {1} and {2}.";
+
+            var expectedProperties = new object[] { "0", "100" };
+            var expectedMessage = "The field Length must be between 0 and 100.";
+
+            var stringLocalizer = new Mock<IStringLocalizer>();
+            stringLocalizer.Setup(s => s[attribute.ErrorMessage, expectedProperties])
+                .Returns(new LocalizedString(attribute.ErrorMessage, expectedMessage));
```

Comment:
It is the job of an IStringLocalizer to insert the values into their places in the format string, so it is expected and necessary that the return value of our Mock here is the fully formed and substituted string. Unless that's not what you're saying?

There IS however the problem there that we haven't actually passed the stringLocalizer.Object into the adapter, which I'll clean up here and anywhere else I spot it.